overpass/api.py

```diff
@@ -191,21 +192,49 @@ def _as_geojson(self, elements):
         geometry = None
         for elem in elements:
             elem_type = elem.get("type")
-            if elem_type and elem_type == "node":
+            elem_tags = elem.get("tags")
+            elem_geom = elem.get("geometry", [])
+            if elem_type == "node":
+                # Create Point geometry
                 geometry = geojson.Point((elem.get("lon"), elem.get("lat")))
-            elif elem_type and elem_type == "way":
-                points = []
-                geom = elem.get("geometry")
-                if geom:
-                    for coords in elem.get("geometry"):
-                        points.append((coords["lon"], coords["lat"]))
-                    geometry = geojson.LineString(points)
+            elif elem_type == "way":
+                # Create LineString geometry
+                geometry = geojson.LineString([(coords["lon"], coords["lat"]) for coords in elem_geom])
+            elif elem_type == "relation":
+                # Initialize polygon list
+                polygons = []
+                # First obtain the outer polygons
+                for member in elem.get("members", []):
+                    if member["role"] == "outer":
+                        points = [(coords["lon"], coords["lat"]) for coords in member.get('geometry', [])]
+                        # Check that the outer polygon is complete
+                        if points and points[-1] == points[0]:
+                            polygons.append([points])
+                # Then get the inner polygons
+                for member in elem.get("members", []):
+                    if member["role"] == "inner":
+                        points = [(coords["lon"], coords["lat"]) for coords in member.get('geometry', [])]
+                        # Check that the inner polygon is complete
+                        if points and points[-1] == points[0]:
+                            # We need to check to which outer polygon the inner polygon belongs
+                            point = Point(points[0])
+                            for poly in polygons:
+                                polygon = Polygon(poly[0])
+                                if polygon.contains(point):
```

Comment:
What if the inner polygon can't be matched with one of the outer? else/finally clause missing